packages/phone-number-privacy/src/salt-generation/bls-cryptography-client.ts

```diff
@@ -0,0 +1,35 @@
+import { AzureKeyVaultClient } from '@celo/contractkit/src/utils/azure-key-vault-client'
+import { BLINDBLS } from 'bls12377js-blind'
+import config from '../config'
+
+export class BLSCryptographyClient {
+  /*
+   * Computes the BLS Salt for the blinded phone number.
+   */
+  public static async computeBLSSalt(queryPhoneNumber: string) {
+    try {
+      const privateKey = await BLSCryptographyClient.getPrivateKey()
+      const pkBuff = new Buffer(privateKey)
+      return BLINDBLS.computePRF(pkBuff, new Buffer(queryPhoneNumber))
+    } catch (e) {
+      console.error('Failed to compute salt', e)
+      throw e
+    }
+  }
+
+  private static privateKey: string
+
+  /**
+   * Get singleton privateKey
+   */
+  private static async getPrivateKey(): Promise<string> {
+    if (BLSCryptographyClient.privateKey) {
+      return BLSCryptographyClient.privateKey
```

Comment:
when would `BLSCryptographyClient.privateKey` already be set?  i don't see it being set anywhere in this diff

Reply:
`getPrivateKey()` can be called multiple times and it doesn't make sense to make a round-trip network request just to get the same key. This just checks if it's already populated the cached value and returns it if it's present.

Reply:
ya but I don't see you ever populating the cache value, so this would always be null

Reply:
you're totally right, thanks!